File: src-all/Python/S0249-group-shifted-strings.py
"""
Given a string, we can &quot;shift&quot; each of its letter to its successive letter, for example: &quot;abc&quot; -> &quot;bcd&quot;. We can keep &quot;shifting&quot; which forms the sequence:

&quot;abc&quot; -> &quot;bcd&quot; -> ... -> &quot;xyz&quot;

Given a list of non-empty strings which contains only lowercase alphabets, group all strings that belong to the same shifting sequence.

Example:

Input: [&quot;abc&quot;, &quot;bcd&quot;, &quot;acef&quot;, &quot;xyz&quot;, &quot;az&quot;, &quot;ba&quot;, &quot;a&quot;, &quot;z&quot;],
Output: 
[
  [&quot;abc&quot;,&quot;bcd&quot;,&quot;xyz&quot;],
  [&quot;az&quot;,&quot;ba&quot;],
  [&quot;acef&quot;],
  [&quot;a&quot;,&quot;z&quot;]
]


"""
from typing import List
class Solution:
    def groupStrings(self, strings: List[str]) -> List[List[str]]:
        
        pass


if __name__ == '__main__':
    assert Solution().groupStrings(0) == 0

File: src-all/Rust/S0249-group-shifted-strings.rs
/*
Given a string, we can "shift" each of its letter to its successive letter, for example: "abc" -> "bcd". We can keep "shifting" which forms the sequence:

"abc" -> "bcd" -> ... -> "xyz"

Given a list of non-empty strings which contains only lowercase alphabets, group all strings that belong to the same shifting sequence.

Example:

Input: ["abc", "bcd", "acef", "xyz", "az", "ba", "a", "z"],
Output: 
[
  ["abc","bcd","xyz"],
  ["az","ba"],
  ["acef"],
  ["a","z"]
]


*/
pub struct Solution {}
impl Solution {
    pub fn group_strings(strings: Vec<String>) -> Vec<Vec<String>> {
        
    }
}

fn main() {
  assert_eq!(0, Solution::group_strings(0));
  println!("Pass test cases!");
}
